tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -2707,7 +2707,7 @@ std::vector<StringRef> CGMSHLSLRuntime::ParseSubobjectExportsAssociations(String
     parsedExports.emplace_back(StringRef(pLast, pData - pLast));
   }
   
-  return std::move(parsedExports);
+  return parsedExports;
```

Comment:
return std::move(...) stops return value optimization. RVO is faster than move, according to a report from IBM C++ compiler team (https://www.ibm.com/developerworks/community/blogs/5894415f-be62-4bc0-81c5-3956e82276f3/entry/RVO_V_S_std_move?lang=en).